--- ./RainbowParenthsis.vim	2010-10-30 20:36:20.000000000 +0600
+++ /home/maks/.shellrc/config/soft/vim_old/vim/plugin/RainbowParenthsis.vim	2010-07-24 13:21:52.000000000 +0600
@@ -11,6 +11,9 @@
 
 " This file is public domain.
 
+if &ft == "vimwiki"
+	finish
+endif
 
 " define colors. Note that the one numbered '16' is the outermost pair,
 " keep that in mind if you want to change the colors.
@@ -18,23 +21,23 @@
 " guifg=xx or ever termfg=, though what good this script will do on a 
 " black and white terminal I don't know.
 hi level1c ctermfg=brown
-hi level2c ctermfg=Darkblue
-hi level3c ctermfg=darkgray
+hi level2c ctermfg=76
+hi level3c ctermfg=202
 hi level4c ctermfg=darkgreen
-hi level5c ctermfg=darkcyan
+hi level5c ctermfg=45
 hi level6c ctermfg=darkred
 hi level7c ctermfg=darkmagenta
 hi level8c ctermfg=brown
-hi level9c ctermfg=gray
-hi level10c ctermfg=black
+hi level9c ctermfg=3
+hi level10c ctermfg=100
 hi level11c ctermfg=darkmagenta
-hi level12c ctermfg=Darkblue
-hi level13c ctermfg=darkgreen
+hi level12c ctermfg=33
+hi level13c ctermfg=46
 hi level14c ctermfg=darkcyan
 hi level15c ctermfg=darkred
-hi level16c ctermfg=red
-
+hi level16c ctermfg=208
 
+""()
 
 " These are the regions for each pair.
 " This could be improved, perhaps, by makeing them match [ and { also,
